keys/signer_secp256k1.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+)
+
+// SignerSecp256k1 is initialized from a keypair
+type SignerSecp256k1 struct {
+	KeyPair *KeyPair
+}
+
+// PublicKey returns the PublicKey of the signer
+func (s SignerSecp256k1) PublicKey() *types.PublicKey {
+	return s.KeyPair.PublicKey
+}
+
+// Signs arbitrary payloads using a KeyPair
+func (s SignerSecp256k1) Sign(payload *types.SigningPayload, sigType types.SignatureType) (*types.Signature, error) {
+	err := s.KeyPair.IsValid()
+	if err != nil {
+		return nil, err
+	}
+	privKeyBytes := s.KeyPair.PrivateKey.Bytes
+
```

Comment:
nit: you should do a similar check here for ensuring `payload.SignatureType == sigType`... I'd recommend making a helper function as this will be pretty common